From 4374791563602f026b870b0c37979506f02683ae Mon Sep 17 00:00:00 2001
From: Tomas Babej <tbabej@redhat.com>
Date: Mon, 6 May 2013 07:23:08 -0400
Subject: [PATCH] Build fixes for Fedora 19

This patch contains various (gcc-related) fixes to make build
on F19 possible.
---
 src/acl.c         | 4 ++--
 src/ldap_helper.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/acl.c b/src/acl.c
index 754cd53dc3d31b99d0954836feafbd46747c48c2..2d4f54db89c73b7e9387f21bbd94eedc364c152b 100644
--- a/src/acl.c
+++ b/src/acl.c
@@ -179,7 +179,7 @@ parse(cfg_parser_t *parser, const char *string, cfg_type_t **type,
 	RUNTIME_CHECK(isc_once_do(&once, init_cfgtypes) == ISC_R_SUCCESS);
 
 	string_len = strlen(string);
-	isc_buffer_init(&buffer, string, string_len);
+	isc_buffer_init(&buffer, (char *)string, string_len);
 	isc_buffer_add(&buffer, string_len);
 
 	result = cfg_parse_buffer(parser, &buffer, *type, &ret);
@@ -296,7 +296,7 @@ get_fixed_name(const cfg_obj_t *obj, const char *name, dns_fixedname_t *fname)
 	str = cfg_obj_asstring(obj);
 
 	len = strlen(str);
-	isc_buffer_init(&buf, str, len);
+	isc_buffer_init(&buf, (char *)str, len);
 
 	/*
 	 * Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=728925
diff --git a/src/ldap_helper.c b/src/ldap_helper.c
index 037629f3e50fdba2c17c1a180736ed3b64169aae..eda17fa8fed63e43d702c7ad9984b176e0b27065 100644
--- a/src/ldap_helper.c
+++ b/src/ldap_helper.c
@@ -2037,7 +2037,7 @@ parse_rdata(isc_mem_t *mctx, ldap_qresult_t *qresult,
 	text.base = rdata_text;
 	text.length = strlen(text.base);
 
-	isc_buffer_init(&lex_buffer, text.base, text.length);
+	isc_buffer_init(&lex_buffer, (char *)text.base, text.length);
 	isc_buffer_add(&lex_buffer, text.length);
 	isc_buffer_setactive(&lex_buffer, text.length);
 
-- 
1.8.1.4

